flow/scenarios/base_scenario.py

```diff
@@ -495,12 +495,15 @@ def gen_random_start_pos(self, initial_config, num_vehicles, **kwargs):
 
         return startpositions, startlanes
 
-    def gen_custom_start_pos(self, initial_config, num_vehicles, **kwargs):
+    @staticmethod
+    def gen_custom_start_pos(cls, initial_config, num_vehicles, **kwargs):
```

Comment:
Question, do you know how class methods fare if parallelism is used? i.e. do all the threads share the class instance?